### PR TITLE
Harry mongo nav tweaks

### PR DESCRIPTION
**Pull request by hswolff**
Check each commit for description of that atomic change. 

- Unify OnElementClick and NavItem enums to one enum

- Revise exports from index.ts and also export NavItem type

- Rename NavItem to NavElement

- When clicking a project menu item close the menu

- move all useState calls in Menu into one location

- Add remaining activeNav values

## ✍️ Proposed changes

🎟 _Jira ticket:_ [Name of ticket](https://jira.mongodb.org/browse/[name-of-ticket])

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

### CODE REVIEW

File: packages/menu/src/Menu.tsx [hunk 1]
```diff
@@ -97,12 +97,21 @@ function Menu({
 
   const hasSetInitialFocus = useRef(false);
   const hasSetInitialOpen = useRef(false);
+  const [focused, setFocused] = useState<HTMLElement>(refs[0] || null);
```
Comment (hswolff) [inline packages/menu/src/Menu.tsx]: Drive by, just moving some code around that increased my readability. functionality is the same

Comment (brookescarlett) [reply packages/menu/src/Menu.tsx]: Should we run a changeset against this, or just wait until we update Menu further to publish?

File: packages/mongo-nav/src/types.ts [hunk 2]
```diff
@@ -1,13 +1,23 @@
-const OnElementClick = {
+const NavElement = {
```
Comment (brookescarlett) [inline packages/mongo-nav/src/types.ts]: Much happier with this combined, and also makes the OnElementClick logic more readable 👍

File: packages/menu/src/Menu.tsx (hunk 1 above)
Comment (hswolff) [reply packages/menu/src/Menu.tsx]: This has no functional change at all, so can just wiat.

**Pull request merged**